cirq-core/cirq/sim/simulator_base.py

```diff
@@ -291,12 +291,19 @@ def _run(
             for k, r in step_result._classical_data.records.items():
                 if k not in records:
                     records[k] = []
-                records[k].append(r)
+                records[k].append(np.array(r, dtype=np.uint8))
             for k, cr in step_result._classical_data.channel_records.items():
                 if k not in records:
                     records[k] = []
-                records[k].append([cr])
-        return {str(k): np.array(v, dtype=np.uint8) for k, v in records.items()}
+                records[k].append(np.array([cr], dtype=np.uint8))
```

Comment:
Nit: Thoughts on `asarray` here ?

Reply:
Actually probably best not to convert to np type at all yet here since they're thrown away by pad_evenly. Then the input type to pad_evenly will be `Sequence[Sequence[Sequence[int]]]`